localstack/contrib/thundra.py

```diff
@@ -270,6 +280,111 @@ def _prepare_invocation_for_node_lambda(context: InvocationContext) -> Additiona
+        return versions[0]
+    except Exception as e:
+        print("Unable to get latest version of Thundra Python agent: %s" % e)
+        return None
+
+
+def _init_python_agent_configs() -> bool:
+    global THUNDRA_PYTHON_AGENT_VERSION
+    global THUNDRA_PYTHON_AGENT_LOCAL_PATH
+
+    latest_version = _get_latest_python_agent_version()
+    version = os.getenv("THUNDRA_AGENT_PYTHON_VERSION", latest_version)
+    if not version:
+        return False
+
+    THUNDRA_PYTHON_AGENT_VERSION = version.strip()
+    THUNDRA_PYTHON_AGENT_LOCAL_PATH = "%s/thundra/python/%s/" % (
+        config.TMP_FOLDER,
+        THUNDRA_PYTHON_AGENT_VERSION,
+    )
+
+    return True
+
+
+def _install_python_agent() -> bool:
+    # Install Thundra Python agent PIP package
+    if not os.path.exists(THUNDRA_PYTHON_AGENT_LOCAL_PATH):
+        install.log_install_msg("Thundra Python agent", verbatim=True)
+        try:
+            install_thundra_cmd = "pip install --target=%s thundra==%s --no-warn-conflicts" % (
+                THUNDRA_PYTHON_AGENT_LOCAL_PATH,
+                THUNDRA_PYTHON_AGENT_VERSION,
+            )
+            subprocess.check_output(install_thundra_cmd, shell=True).decode()
+        except Exception as e:
+            print("Unable to install Thundra Python agent: %s" % e.output.decode())
+            return False
+    return True
+
+
+def _is_python_lambda_with_support_version(func_details):
+    runtime = getattr(func_details, "runtime", func_details)
+    return runtime in [
+        LAMBDA_RUNTIME_PYTHON36,
+        LAMBDA_RUNTIME_PYTHON37,
+        LAMBDA_RUNTIME_PYTHON38,
+    ]
+
+
+def _prepare_invocation_for_python_lambda(
+    context: InvocationContext,
+) -> AdditionalInvocationOptions:
+    # Download and initialize Python agent
+    _ensure_python_agent_initialized()
+
+    # If agent could not be initialized, skip here
+    if not THUNDRA_PYTHON_AGENT_INITIALIZED:
+        return None
+
+    result = AdditionalInvocationOptions()
+
+    # Make sure API key is contained in environment
+    result.env_updates[THUNDRA_APIKEY_ENV_VAR_NAME] = _get_apikey(context.environment)
+
+    # Switch handler to Thundra and pass original handler to Thundra through environment variable
+    result.env_updates[THUNDRA_AGENT_LAMBDA_HANDLER_ENV_VAR_NAME] = context.handler
+    result.updated_handler = "thundra.handler.wrapper"
+
+    # If log disable is not configured explicitly, set it to false to enable log capturing by default
+    log_disabled = context.environment.get(THUNDRA_AGENT_LOG_DISABLE_ENV_VAR_NAME)
+    if not log_disabled:
+        result.env_updates[THUNDRA_AGENT_LOG_DISABLE_ENV_VAR_NAME] = "false"
+
+    # Map Thundra agent path into container so it will be accessible by Lambda function Python environment
+    agent_path_mapping = "-v %s/:/opt/python/" % THUNDRA_PYTHON_AGENT_LOCAL_PATH
+
+    if context.docker_flags:
+        context.docker_flags = f"{context.docker_flags} {agent_path_mapping}"
```

Comment:
Note to self: I think this is good for now, but over time we should look into adding `additional_docker_flags` to `AdditionalInvocationOptions`, making the incoming `context` immutable... (to avoid that plugin implementations overwrite each other's Docker flags / configs..)

Reply:
Should we handle it in this PR?

Reply:
No, not required - we'll pick that up in a future PR.. (was more of a note to self.. ;) ). thx